mvpa2/tests/test_giftidataset.py

```diff
@@ -0,0 +1,185 @@
+        samples = ["SgwCQB1aZL+8dFO/L92UPw==",
+                   "BoEVPzEI7D99P5U/hxZZvw==",
+                   "/tS4v90khr5U4/W/pHBFQA==",
+                   "ppsEvy2yFUDByuE+aJHtPw==",
+                   "sHKYP2DlUL4ZBFa+16PgPw==",
+                   "XrqpvwRWLkBQjRc+EoMAPw=="]
+        nodes = "CAAAAAcAAAACAAAAAwAAAA=="
+
+    elif format_ == 'GZipBase64Binary':
+        samples = ["eJzz4mFykI1K2b+nJHi//t0p9gAsDAZU",
+                   "eJxjaxS1N+R4Y19rP9W+XSxyPwAolAWF",
+                   "eJz7d2XH/rsqbftCHn/dv6TA1QEAXukKEw==",
+                   "eJxbNptlv+4mUYeDpx7aZUx8aw8AQboICA==",
+                   "eJzbUDTDPuFpwD5JlrB91xc/sAcAQ0gIFw==",
+                   "eJyL27VyP0uYnkNAr7idUDODPQA14AVP"]
+        nodes = "eJzjYGBgYAdiJiBmBmIAAQAAFQ=="
+
+    nfeatures = 4
+
+    return samples, nodes, nfeatures
+
+
+
+def _get_test_dataset(include_nodes=True):
+    # returns test dataset matching the contents of _get_test_sample_node_data
+    samples, nodes, _ = _get_test_sample_node_data()
+    ds = Dataset(np.asarray(samples))
+
+    if include_nodes:
+        ds.fa['node_indices'] = np.asarray(nodes)
+
+    nsamples = ds.nsamples
+    ds.sa['intents'] = ['NIFTI_INTENT_NONE'] * nsamples
+
+    return ds
+
+
+
+def _build_gifti_string(format_, include_nodes=True):
+    # builds the string contents of a GIFTI file
+    samples, nodes, nfeatures = _get_test_sample_node_data(format_)
+    nsamples = len(samples)
+
+    ndata_arrays = nsamples + (1 if include_nodes else 0)
+
+    header = """<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE GIFTI SYSTEM "http://www.nitrc.org/frs/download.php/115/gifti.dtd">
+<GIFTI Version="1.0"  NumberOfDataArrays="%d">""" % (ndata_arrays)
+
+
+    def _build_data_array(data, intent, encoding=format_, dim=nfeatures):
+        datatype = "INT32" if intent == "NODE_INDEX" else "FLOAT32"
+
+        return """<MetaData/>
+                   <LabelTable/>
+                   <DataArray  ArrayIndexingOrder="RowMajorOrder"
+                               DataType="NIFTI_TYPE_%s"
+                               Dim0="%d"
+                               Dimensionality="1"
+                               Encoding="%s"
+                               Endian="LittleEndian"
+                               ExternalFileName=""
+                               ExternalFileOffset=""
+                               Intent="NIFTI_INTENT_%s">
+                      <MetaData>
+                      </MetaData>
+                      <Data>%s</Data>
+                    </DataArray>
+                """ % (datatype, dim, encoding, intent, data)
+
+
+    body_elements = []
+    if include_nodes:
+        body_elements.append(_build_data_array(nodes, 'NODE_INDEX'))
+
+    body_elements.extend(_build_data_array(sample, 'NONE')
+                         for sample in samples)
+
+    footer = """</GIFTI>"""
+
+    return header + "".join(body_elements) + footer
+
+
+
+@with_tempfile(suffix='.func.gii')
+def test_gifti_dataset(fn):
```

Comment:
How about adding some h5save/h5load to test_giftidataset? That way I don't have to duplicate code to generate the surface-based datasets.